reporter/symbol_uploader.go

```diff
@@ -62,58 +62,65 @@ type uploadData struct {
 }
 
 type DatadogSymbolUploader struct {
-	ddAPIKey             string
-	ddAPPKey             string
-	intakeURL            string
+	ddAPIKeys            []string
+	ddAPPKeys            []string
+	intakeURLs           []string
 	version              string
 	dryRun               bool
 	uploadDynamicSymbols bool
 	uploadGoPCLnTab      bool
 	workerCount          int
 
-	uploadCache   *lru.SyncedLRU[libpf.FileID, struct{}]
-	client        *http.Client
-	uploadQueue   chan uploadData
-	symbolQuerier *DatadogSymbolQuerier
+	uploadCache    *lru.SyncedLRU[libpf.FileID, struct{}]
+	client         *http.Client
+	uploadQueue    chan uploadData
+	symbolQueriers []*DatadogSymbolQuerier
 }
 
-func NewDatadogSymbolUploader(cfg SymbolUploaderConfig) (*DatadogSymbolUploader, error) {
+func NewDatadogSymbolUploader(cfg *SymbolUploaderConfig) (*DatadogSymbolUploader, error) {
 	err := exec.Command("objcopy", "--version").Run()
 	if err != nil {
 		return nil, fmt.Errorf("objcopy is not available: %w", err)
 	}
 
-	if cfg.APIKey == "" {
-		return nil, errors.New("API key is not set")
+	if cfg.APIKeys == nil {
+		return nil, errors.New("no API key is set")
 	}
 
-	if cfg.APPKey == "" {
-		return nil, errors.New("application key is not set")
+	if cfg.APPKeys == nil {
+		return nil, errors.New("no application key is set")
 	}
 
-	if cfg.Site == "" {
-		return nil, errors.New("site is not set")
+	if cfg.Sites == nil {
+		return nil, errors.New("no site is set")
```

Comment:
Validate than len(symbolEndpoints) != 0 (isn't semantically the same as comparing to `nil`)